scripts/valhalla_build_config

```diff
@@ -534,54 +535,65 @@ help_text = {
-    parser.add_argument(arg, type = t, help=h, default=tree)
+def add_leaf_args(path: str, tree: Union[dict, bool, str, list, Optional], leaves_: List[str], parser_: ArgumentParser, help: dict):
+    """
+    returns a list of leaves of the tree, `\0` separated, stops at non dicts
+    while doing so it also adds arguments to the parser
+    """
+    # if we are at a dict go deeper
+    if isinstance(tree, dict):
+        for k in tree:
+            v = tree[k]
+            add_leaf_args('\0'.join([path, k]) if len(path) else k, v, leaves_, parser_, help)
+    # we've reached a leaf
+    else:
+        keys = path.split('\0')
+        h = help
+        for k in keys:
+            h = h[k]
 
-    leaves.append(path)
+        # its either required and is the right type or optional and has a type to use
+        # lists are supported as comma separated and bools support a bunch of conventions
+        # the rest of the primatives (strings and numbers) parse automatically
+        if isinstance(tree, Optional):
+          t = tree.type
+        elif isinstance(tree, list):
+          t = lambda arg: arg.split(',')
+        elif isinstance(tree, bool):
+          t = lambda arg: bool(strtobool(arg))
+        else:
+          t = type(tree)
 
-#entry point to program
-if __name__ == '__main__':
+        arg = '--' + path.replace('_', '-').replace('\0', '-')
+        parser_.add_argument(arg, type=t, help=h, default=tree)
+        leaves_.append(path)
 
-  #set up program options
-  leaves = []
-  parser = argparse.ArgumentParser(description='Generate valhalla configuration', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-  add_leaf_args('', config, leaves, parser)
-  #TODO: add argument to set base path and use in all other path based values
-  args = parser.parse_args()
 
-  #override the defaults given what was passed
-  for leaf in leaves:
-    keys = leaf.split('\0')
-    v = config
-    for k in keys[:-1]:
-      v = v[k]
-    v[keys[-1]] = getattr(args, leaf.replace('\0', '_'))
-    if isinstance(v[keys[-1]], type(optional())):
-      del v[keys[-1]]
+def override_config(args_: dict, leaves_: list, config_: dict):
+    """override the defaults given what was passed"""
+    for leaf in leaves_:
+        keys = leaf.split('\0')
+        v = config_
+        for k in keys[:-1]:
+            v = v[k]
+        v[keys[-1]] = args_.get(leaf.replace('\0', '_'))
+        if isinstance(v[keys[-1]], type(Optional())):
+            del v[keys[-1]]
+
+
+# set up available program options
+leaves = []
+parser = ArgumentParser(description='Generate valhalla configuration', formatter_class=ArgumentDefaultsHelpFormatter)
+add_leaf_args('', config, leaves, parser, help_text)
+
+# entry point to program
+if __name__ == '__main__':
+    # TODO: add argument to set base path and use in all other path based values
+    args = parser.parse_args()
+
+    # encapsulate to test easier
+    override_config(args.__dict__, leaves, config)
```

Comment:
`args` is a argparser.Namespace, `__dict__` contains all k, v of the arguments and nothing else. again, easier to test that function this way